app/Actions/Place/Checkin/SetCheckinAction.php

```diff
@@ -4,26 +4,34 @@
 
 use Hedonist\Actions\Place\Checkin\Criteria\UserIdAndPlaceIdCriteria;
 use Hedonist\Exceptions\Place\PlaceNotFoundException;
+use Hedonist\Exceptions\User\UserNotFoundException;
 use Hedonist\Repositories\Place\CheckinRepositoryInterface;
 use Hedonist\Entities\Place\Checkin;
 use Hedonist\Repositories\Place\PlaceRepositoryInterface;
+use Illuminate\Support\Facades\Auth;
 
 class SetCheckinAction
 {
     protected $repository;
     protected $placeRepository;
+    private $userId;
 
     public function __construct(CheckinRepositoryInterface $repository, PlaceRepositoryInterface $placeRepository)
     {
         $this->repository = $repository;
         $this->placeRepository = $placeRepository;
+        if (Auth::check()) {
```

Comment:
No need for Auth::check() , this SetCheckinAction is protected and only authenticated user can request it.